sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQuerySource.java

```diff
@@ -76,11 +69,12 @@
   private final ValueProvider<String> query;
   private final Boolean flattenResults;
   private final Boolean useLegacySql;
-  private transient AtomicReference<JobStatistics> dryRunJobStats;
   private final QueryPriority priority;
   private final String location;
   private final String kmsKey;
 
+  private transient AtomicReference<BigQueryQueryHelper> queryHelperReference;
```

Comment:
I've refactored this so that BigQueryQueryHelper is just a collection of static helper methods, and all the state -- including the transient state for the results of the query dry run -- are members of the source object, as before.